packages/wallet/api/src/lib-wallet.js

```diff
@@ -1790,6 +1791,29 @@ export function makeWalletRoot({
     serialize: context.serialize,
   });
 
+  const handleAcceptOfferAction = async offer => {
+    const { requestContext } = offer;
+    const rawId = await addOffer(offer, requestContext);
+    const dappOrigin =
+      requestContext.dappOrigin || requestContext.origin || 'unknown';
+    return acceptOffer(`${dappOrigin}#${rawId}`);
+  };
+
+  const handleSuggestIssuerAction = ({ petname, boardId }) =>
+    suggestIssuer(petname, boardId);
+
+  const performAction = async obj => {
+    const { type, data } = JSON.parse(obj.action);
+    switch (type) {
+      case 'acceptOffer':
+        return handleAcceptOfferAction(data);
+      case 'suggestIssuer':
+        return handleSuggestIssuerAction(data);
+      default:
+        throw new Error(`Unknown wallet action ${type}`);
+    }
+  };
```

Comment:
This pattern requires naming things twice. consider,
```js
const handlers = {
  async acceptOffer(offer) {
  …
  },
  suggestIssuer({ petname, boardId }) {
    return suggestIssuer(petname, boardId);
  }
};
const performAction = obj => {
  const { type, data } = JSON.parse(obj.action);
  const handle = handlers[type];
  if (!handle) {
    throw new Error(`Unknown wallet action ${type}`);
  }
  return handle(data);
```

Also note `performAction` need not by synchronous. It does need to return a promise. Giving it jsdoc can make that clear.